shortcodes/class.geojson-shortcode.php

```diff
@@ -115,26 +115,28 @@ function layerStyle (feature) {
                 function onEachFeature (feature, layer) {
                     var props = feature.properties || {},
                         text = popup_property && props[ popup_property ] || template(popup_text, feature.properties),
-                        species = feature.properties.name,
+                        featureName = feature.properties.name,
                         iconUrl = feature.properties.iconUrl;
                     if (text) {
                         layer.bindPopup( text );
                     }
-                    // create the legend entities for the species names, assumes the filename pattern is *-[iconColor].png
-                    legendVals[species] = iconUrl.slice(iconUrl.lastIndexOf("-")+1, iconUrl.length-4);
+                    if (iconUrl) {
+                        // create the legend entities for the feature names, assumes the filename pattern is *-[iconColor].*
+                        legendVals[featureName] = iconUrl.slice(iconUrl.lastIndexOf("-")+1, iconUrl.lastIndexOf("."));
```

Comment:
Best to put spaces around operators, i.e. `lastIndexOf("-") + 1`

See https://www.w3schools.com/js/js_conventions.asp

Reply:
Oops, thanks for the reminder.